failure-summary.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Finds failing Checks and combines each help message into an array
+ * @param  {Object} nodeData Individual "detail" object to generate help messages for
+ * @return {String}          failure messages
+ */
+var failureSummary = function failureSummary(nodeData) {
+	'use strict';
+
+	var failingChecks = {};
+	// combine "all" and "none" as messaging is the same
+	failingChecks.none = nodeData.none.concat(nodeData.all);
+	failingChecks.any = nodeData.any;
+
+	return Object.keys(failingChecks).map(function (key) {
```

Comment:
failureSummary is called for both "any" and "none". then failureSummary will call failureMessage which I copied from https://github.com/dequelabs/axe-core/blob/master/lib/core/reporters/helpers/failure-summary.js

thats where the message come from.